src/os/ui/servers.js

```diff
@@ -369,7 +369,13 @@ os.ui.ServersCtrl.prototype.remove = function(provider, opt_prompt) {
   }
 
   os.settings.delete(['userProviders', provider.getId()]);
-
+  var descList = os.dataManager.getDescriptors();
+  for (var i = 0, ii = descList.length; i < ii; i++) {
+    var desc = descList[i];
+    if (desc.getDataProvider() == provider) {
+      os.dataManager.removeDescriptor(desc);
+    }
+  }
```

Comment:
This block is incorrect. The value provided by `descriptor.getProvider()` is only intended for display. Rather, you should remove descriptors via ID like so:

```
var descList = os.dataManager.getDescriptors(provider.getId() + os.data.BaseProvider.ID_DELIMITER);
for (var i = 0, ii = descList.length; i < ii; i++) {
  os.dataManager.removeDescriptor(descList[i]);
}
```